perllib/FixMyStreet/App/Controller/Tokens.pm

```diff
@@ -66,10 +66,8 @@ sub confirm_problem : Path('/P') {
     # For Zurich, email confirmation simply sets a flag, it does not change the
     # problem state, log in, or anything else
     if ($c->cobrand->moniker eq 'zurich') {
-        my $extra = { %{ $problem->extra || {} } };
-        $extra->{email_confirmed} = 1;
+        $problem->set_extra_metadata( $c, email_confirmed => 1 );
```

Comment:
Presumably this `$c` should not be here. Please include a test along with the fix for this to show that this code is being tested.